pkg/autoscaler/metrics/http_scrape_client.go

```diff
@@ -107,28 +108,26 @@ func (c *httpScrapeClient) Scrape(url string) (Stat, error) {
 		return emptyStat, fmt.Errorf("GET request for URL %q returned HTTP status %v", url, resp.StatusCode)
 	}
 	if resp.Header.Get("Content-Type") == network.ProtoAcceptContent {
-		return statFromProto(resp.Body, resp.ContentLength)
+		length, err := strconv.Atoi(resp.Header.Get("Content-Length"))
+		if err != nil {
+			return emptyStat, fmt.Errorf("Parsing Content-Length failed.")
+		}
```

Comment:
Are these necessary? Doesn't the response auto-parse this into the ContentLength field? Sorry if there was a misunderstanding here :sweat:

Reply:
I didnt see that part in the http client but will double check.

Reply:
@markusthoemmes this is confusing, In the following simple case:
```
	server := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
		w.Header().Add("Content-Type", "application/protobuf")

		buf, _ := stat.Marshal()
		if _, err := w.Write(buf); err != nil {
			b.Error(err)
		}
	}))
```
the server sets th elength and the resp.ConteLength is set by the transport roundtriper at the other end. In our tests we use a fake one that is why I needed to set it explicitly. Then I saw some integration tests failing and setting up `Content Length` explicitly fixed it. So I suspect in these tests clients something is not setup correctly.

Reply:
The HTTP implementation can't always handle that because the content-length isn't necessarily known at the time the headers are written. In theory, you can continue writing to the body forever even. I think Golang sets it for rather small bodys, if the handlers quit quickly, as I said though, I'd err on the side of caution and manually set it (as we do now).

Reply:
Ok I can test if `resp.ContentLength` is set correctly and if not extract the value manually, is that what you mean?

Reply:
I believe `resp.ContentLength` is always just parsed for you already (see https://golang.org/src/net/http/transfer.go, line 510 and on). I don't think we need to extract the value manually.

We should set it manually though (which is what we do I believe now), in the server, to make sure it's always correctly set.